src/dune_rules/gen_rules.ml

```diff
@@ -386,35 +386,41 @@ type for_melange =
     stanza : Melange_stanzas.Emit.t
   }
 
-let rec nearest_parent_melange_emit ~dir =
+let rec nearest_parent_melange_emits ~dir =
```

Comment:
Can you revert to the original approach with `under_melange_emit_target`? I think it's confusing to collect this list of emit stanzas when only one is going to be relevant.

Reply:
I can do that, but I think that'll imply we can't reuse this function.

Reply:
I don't see why not, in both uses of this function, you end up looking at only one element in the list.

Reply:
yeah, my bad, we can reuse the logic.

Reply:
I think I had just originally overcomplicated the logic here. `a/ (target output)` and `a/b/c/ (target output)` are allowed to co-exist, and I was being overly restrictive. However `a/ (target output)` and `a/output/b` can't, and the original function can detect this case too.

Reply:
This is fixed in the most recent push.